### ISSUE TITLE
How to get the init_prototypes.pt and labels_14.pickle?

### ISSUE THREAD

**Issue by Zhang1Sheng** [Markin-Wang/XProNet#4]
Dear author, I am honored to read your excellent papers and codes. But I have some questions to ask you. I want to modify the generation method of .pt and .pickle files, but this part of code doesn't appear in the project, so could you please send me this part of code to help me refer to it?Thank you very much.
My email address is 1434012475@qq.com

**Comment by Markin-Wang** [comment]
Hi, thanks for your interest. You can refer to our paper to generate the labels and initialize the prototype. Generally speaking, you firstly need to use the pretrained models to extract the visual and textual features. Then, k-mean is utlized to cluster the features. I added the scripts to generate the image prototype and also the labels (use CheXpert to get the labels, and texual protoype, not included in this repo) on "file" directory. Note that the code of this part is not well-organized ( we will organize it in the future ), so you may need to make some modifications.

**Comment by Zhang1Sheng** [comment]
Excuse me, sir，what is the training set of CheXbert model? Is it the training set used in CheXbert's paper? Did you directly use iu-xray and mimic-cxr as test sets of CheXbert model to generate pseudo-labels?

**Comment by Markin-Wang** [comment]
> Excuse me, sir，what is the training set of CheXbert model? Is it the training set used in CheXbert's paper? Did you directly use iu-xray and mimic-cxr as test sets of CheXbert model to generate pseudo-labels?

Hi, as mentioned in our paper, we directly use the pretrained CheXbert provided by the authors to generate the prototypes/labels based on the reports. Hence, there is no need to train the CheXbert model again.

**Comment by Zhang1Sheng** [comment]
Excuse me, sir, I have another question, that is, in the results generated by CheXbert model, each class has four values: "negative", "positive", "uncertain" and "blank". How do you convert them to 0 and 1?Could you please explain it in detail?Thank
 you very much.

**Comment by Markin-Wang** [comment]
> uncertain

We consider "positive" and "uncertain" as positive (1), and "negative"  and "uncertain" as negative (0).

**Comment by Markin-Wang** [comment]
> Excuse me, sir, I have another question, that is, in the results generated by CheXbert model, each class has four values: "negative", "positive", "uncertain" and "blank". How do you convert them to 0 and 1?Could you please explain it in detail?Thank you very much.

We consider "positive" and "uncertain" as positive (1), and "negative" and "uncertain" as negative (0).

**Comment by Zhang1Sheng** [comment]
> > Excuse me, sir, I have another question, that is, in the results generated by CheXbert model, each class has four values: "negative", "positive", "uncertain" and "blank". How do you convert them to 0 and 1?Could you please explain it in detail?Thank you very much.
> 
> We consider "positive" and "uncertain" as positive (1), and "negative" and "uncertain" as negative (0).

So what about "blank"?

**Comment by Markin-Wang** [comment]
> > > Excuse me, sir, I have another question, that is, in the results generated by CheXbert model, each class has four values: "negative", "positive", "uncertain" and "blank". How do you convert them to 0 and 1?Could you please explain it in detail?Thank you very much.
> > 
> > 
> > We consider "positive" and "uncertain" as positive (1), and "negative" and "uncertain" as negative (0).
> 
> So what about "blank"?

be 0.
